### ISSUE TITLE
ionization is broken

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#4611]
@pordyna reported:

```
Dear PIConGPU users, it looks like my simulations are freezing because of this PR, when ionization is enabled. Not exactly sure how this could have break anything, since I'm not even using this new manipulator. But anyway, if you use ionization, maybe keep using the dev from before this merge until we can figure it out.
```

![image](https://github.com/ComputationalRadiationPhysics/picongpu/assets/4037125/c60b6910-fb67-495c-a341-6adee4a039d6)

**Comment by psychocoderHPC** [comment]
Maybe this is the reason https://github.com/ComputationalRadiationPhysics/picongpu/pull/4592/files#r1236552772

Can it be that we ionize again and again because bound electrons is not updated? Not sure if we should create many particles until we run out of memory or if we somewhere moves into an endless loop.

**Comment by BrianMarre** [comment]
yes, you are right, I forgot to update the ionization kernels

**Comment by BrianMarre** [comment]
essentially the charge state was never updated by ionization, causing picongpu to repeat the ionization every time step and spawn new macro particles, until it runs out of memory/mallocmc slows to a crawl

**Comment by BrianMarre** [comment]
@psychocoderHPC Issue can be closed, bug is fixed

**Comment by psychocoderHPC** [comment]
fixed with #4615 and #4612
